src/extern.jl

```diff
@@ -12,12 +12,13 @@ Estimate probabilistic PCA model for noise that is heteroscedastic across sample
 + `iters`  : number of iterations to run
 + `init`   : initial model (will be modified in-place)
 + `vknown` : variances are known (do not update) default `false`
++ `varfloor` : lower bound for variance iterates (useful if the iterates are degenerating to zero) default = 0
 Output is a [`HePPCATModel`](@ref) object.
 """
-function heppcat(Y,k,iters::Integer;init=homppca(Y,k),vknown::Bool=false)
+function heppcat(Y,k,iters::Integer;init=homppca(Y,k),vknown::Bool=false,varfloor=zero(eltype(first(Y))))
     M = init
```

Comment:
Setup methods in advance:
```suggestion
    M = init
    vmethod = iszero(varfloor) ? ExpectationMaximization() : ProjectedVariance(ExpectationMaximization(),varfloor)
    Fmethod = ExpectationMaximization()
```